front/lib/api/assistant/actions/websearch.ts

```diff
@@ -293,12 +295,25 @@ export class WebsearchConfigurationServerRunner extends BaseActionConfigurationS
             };
             return;
           }
-          output = outputValidation.right;
+          const rawResults = outputValidation.right.results;
+          const formattedResults: WebsearchResultType[] = [];
+          if (rawResults) {
+            const refs = getRefs().slice(0, rawResults.length ?? 0);
```

Comment:
This will clash with the retrival actions at previous or same step.

@fontanierh introduced a system to get a subspace of references for each action which needs to be reused here?